src/ws_transport.ts

```diff
@@ -52,29 +58,61 @@ export class WsTransport implements Transport {
     this.done = false;
     this.socketClosed = false;
     this.encrypted = false;
+    this.peeked = false;
   }
 
-  async connect(
+  connect(
     server: Server,
     options: ConnectionOptions,
   ): Promise<void> {
     const connected = false;
     const connLock = deferred<void>();
 
+    // ws client doesn't support TLS setting
+    if (options.tls) {
```

Comment:
So this library must be used in context of web browsers (that are responsible to do the TLS handshake)? In other words, you have no plan to have users be able to connect directly to NATS server with WS and TLS?

Reply:
the ws_transport is only used in the context of a browser - so that option is rejected - I have to do a similar thing for some of the options in deno which only takes the ca certificate.

Now - with that said, if they specify an empty tls option `{}` - we could reject the connection if it is not encrypted - this would, ie - they didn't connect to a `wss` URL - We only know this if the URL itself says `wss` - which in the browser world is a 'safe' bet.